src/app/components/execution-environments/compute-resource/compute-resource-selection-criteria/compute-resource-selection-criteria.component.ts

```diff
@@ -6,6 +7,7 @@ import { ComputeResourceDto, SoftwarePlatformDto } from 'api-atlas/models';
 import { ExecutionEnvironmentsService } from 'api-atlas/services';
 import { CreateQpuRequestDto, QpuDto, SdkDto } from 'api-nisq/models';
 import { QpuService, SdksService } from 'api-nisq/services';
+import { ChangePageGuard } from '../../../../services/deactivation-guard';
```

Comment:
Just as a note, you might need to enable the guard in the ``app-routing.module.ts`` with "canDeactivate: [ChangePageGuard]"